platform/security/certificate/security-crl-generator/src/test/java/org/codice/ddf/security/crl/generator/CrlGeneratorTest.java

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.crl.generator;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import javax.ws.rs.core.Response;
+import org.apache.cxf.jaxrs.client.WebClient;
+import org.codice.ddf.cxf.client.ClientFactoryFactory;
+import org.codice.ddf.cxf.client.SecureCxfClientFactory;
+import org.codice.ddf.platform.util.properties.PropertiesLoader;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.osgi.service.event.EventAdmin;
+
+public class CrlGeneratorTest {
+  @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private static final String KEYSTORE_PASS_PROPERTY = "${javax.net.ssl.keyStorePassword}";
+
+  private byte[] demEncodedCrl;
+
+  private EventAdmin eventAdmin = mock(EventAdmin.class);
+
+  @Before
+  public void setup() throws Exception {
+    CrlGenerator.issuerEncryptionPropertiesLocation =
+        getClass().getResource("/issuer/encryption.properties").getPath();
+    CrlGenerator.issuerSignaturePropertiesLocation =
+        getClass().getResource("/issuer/signature.properties").getPath();
+    CrlGenerator.serverEncryptionPropertiesLocation =
+        getClass().getResource("/server/encryption.properties").getPath();
+    CrlGenerator.serverSignaturePropertiesLocation =
+        getClass().getResource("/server/signature.properties").getPath();
+
+    CrlGenerator.crlFileLocation = temporaryFolder.getRoot().toString();
+
+    InputStream inputStream =
```

Comment:
✏️ Though it's only unit test code, should probably use try-with-resources here to ensure streams are closed.